schema_based_on_diagram.sql

```diff
@@ -0,0 +1,50 @@
+CREATE TABLE patients (
+    id SERIAL PRIMARY KEY,
+    name varchar(100),
+    date_of_birth date
+);
+
+CREATE TABLE medical_histories (
+    id SERIAL PRIMARY KEY,
+    admitted_at TIMESTAMP,
+    patients_id INT,
+    status varchar(255),
+    FOREIGN KEY(patients_id) REFERENCES patients(id)
+ );
+ 
+CREATE TABLE treatments(
+    id serial PRIMARY KEY, 
+    type VARCHAR(50),
+    name VARCHAR(50)
+    );
+    
+
+ CREATE TABLE invoices(
+    id serial PRIMARY KEY, 
+    total_amount decimal,
+    generated_at TIMESTAMP,
+	payed_at TIMESTAMP,
+	medical_history_id INT,
+	FOREIGN KEY(medical_history_id) REFERENCES medical_histories(id)
+	);
+	
+CREATE TABLE invoice_items(
+    id serial PRIMARY KEY, 
+    unit_price DECIMAL, 
+    quantity INT, 
+    total_price DECIMAL,
+    invoice_id INT REFERENCES invoices(id), 
+    treatment_id INT,
+    FOREIGN KEY (treatment_id) REFERENCES treatments(id),
+    FOREIGN KEY (invoice_id) REFERENCES invoices(id)    
+);
+
+CREATE TABLE medical_histories_has_treatments (
+    medical_history_id int refrences medical_histories(id),
+    treatment_id int refrences treatments(id),
+    );
```

Comment:
- [ ] Kindly add indexes for those two foreign keys(`medical_history_id`, `treatment_id`) for the table `medical_histories_has_treatments` as requested by the previous reviewers.